gensim/test/test_keyedvectors.py

```diff
@@ -104,6 +104,27 @@ def test_most_similar_topn(self):
         predicted = self.vectors.most_similar('war', topn=None)
         self.assertEqual(len(predicted), len(self.vectors.vocab))
 
+    def test_relative_cosine_similarity(self):
+        """Test relative_cosine_similarity returns expected results with an input of a word pair and topn"""
+        wordnet_syn = ['good', 'goodness', 'commodity', 'trade_good', 'full', 'estimable', 'honorable',
+        'respectable', 'beneficial', 'just', 'upright', 'adept', 'expert', 'practiced', 'proficient',
+        'skillful', 'skilful', 'dear', 'near', 'dependable', 'safe', 'secure', 'right', 'ripe', 'well',
+         'effective', 'in_effect', 'in_force', 'serious', 'sound', 'salutary', 'honest', 'undecomposed',
+          'unspoiled', 'unspoilt', 'thoroughly', 'soundly']   # synonyms for "good" as per wordnet
+        cos_sim = []
+        for i in range(len(wordnet_syn)):
+            if wordnet_syn[i] in self.vectors.vocab:
+                cos_sim.append(self.vectors.similarity("good", wordnet_syn[i]))
+        cos_sim = sorted(cos_sim, reverse=True)  # cosine_similarity of "good" with wordnet_syn in decreasing order
+        # computing relative_cosine_similarity of two similar words
+        rcs_wordnet = self.vectors.similarity("good", "nice") / sum(cos_sim[i] for i in range(10))
```

Comment:
I'm not sure what this is calculating. It's kind of like the `relactive_cosine_similarity()` formula, but now with only WordNet synonyms as contributors to the denominator. And, only those synonyms which happen to be in this vector-set. Are all those words in the `euclidean_vectors.bin` test vectors set? As a result, I'm not sure what the following asserts really test. Is this matching something in the paper?

Reply:
Actually, this is the problem which I found while making test.There is not any **claim or any perfect result** in the paper and I can't find any way to confirm on corpus other than `wordnet`, so I think best way will be to compare the `relative_cosine_similarity` of `wordnet` synonyms and `most_similar` ones under a threshold of `0.125`.

Let me give explain the insights of the section _relative_ cosine similarity of the paper_:
1) Construct a set of the top 10 most (cosine) similar words for w1 (called topn in the paper).
2) Calculate a normalized score for each of the words in the topn, by dividing by the sum of the topn cosine similarity scores.

They mostly wanted to know if the most similar word of w1 was a synonym or not, and not a synonym/hypernym etc. They expected that if the most (cosine) similar word is a lot more (cosine) similar than the other words in the topn it is more likely to be a synonym, than if it is only slightly more similar. So this is what the rcs takes into account.
So, they come to conclusion which is the only claim in the paper is that if a **word pair have a rcs greater 0.10 than it is more likely to be an arbitrary pair**.
`0.10` can be used threshold but this result is based on `wordnet` corpus. On a short corpus this result may be more lower. The threshold is nothing but the mean of the cosine_similarities of topn words. So on a short corpus it may be anything less than `0.10`. 

@gojomo ,Can you suggest some better way to test why looking at above description?